deepmd/utils/data_system.py

```diff
@@ -509,6 +518,121 @@ def _prob_sys_size_ext(self, keywords):
         return sys_probs
 
 
+class DeepmdDataDocker() :
+    """
+    Class for manipulating many data dockers. 
+    It is implemented with the help of DeepmdData
+    """
+    def __init__ (self,
+                  data_systems : List[DeepmdDataSystem],
+                  batch_size : int,
+                  type_map : List[str] = None
+                  ) :
+        """
+        Constructor
+        
+        Parameters
+        ----------
+        datasystems
+                Combination of several DeepmdDataSystems
+        batch_size
+                The batch size
+        type_map
+                Gives the name of different atom types
+        """
+        # init data
+        self.batch_size = batch_size
+        self.data_systems = data_systems
+        # natoms, nbatches
+        self.nmethod = len(self.data_systems)
+        self.pick_idx = 0
+        nbatch_list = []
+        batch_size_list=[]
+        name_list = []
+        for ii in range(self.nmethod) :
+            nbatch_list.extend(self.data_systems[ii].get_nbatches())
+            batch_size_list.extend(self.data_systems[ii].get_batch_size())
+            name_list.append(str(self.data_systems[ii].get_name()))
+        self.type_map = type_map
+        self.nbatches = list(nbatch_list)
+        self.batch_size = list(batch_size_list)
+        self.name_list = list(name_list)
+
+
+    def get_nmethod(self):
+        return self.nmethod
+        
+    def get_batch(self, method_idx : int = None,sys_idx : int = None):
+        if method_idx is not None :
+            self.pick_idx = method_idx
+        s_data = self.data_systems[self.pick_idx]
+        b_data = s_data.get_batch(sys_idx)
+        
+        return b_data
+
+    def get_data_system(self,name):
+        for i in range(self.nmethod):
+            if self.name_list[i] == name:
+                return self.data_systems[i]
```

Comment:
Here is a more Pythonic-style code:
```suggestion
        for iname, idata_system in zip(self.name_list, self.data_systems):
            if iname == name:
                return idata_system

```